.github/workflows/create-release.yml

```
@@ -0,0 +1,188 @@
+      # the known GitHub issue described in https://github.com/camunda/camunda/issues/28522
+      - name: Generate GitHub token
+        id: github-token
+        uses: camunda/infra-global-github-actions/generate-github-app-token-from-vault-secrets@main
+        with:
+          github-app-id-vault-key: MONOREPO_RELEASE_APP_ID
+          github-app-id-vault-path: secret/data/products/camunda/ci/camunda
+          github-app-private-key-vault-key: MONOREPO_RELEASE_APP_PRIVATE_KEY
+          github-app-private-key-vault-path: secret/data/products/camunda/ci/camunda
+          vault-auth-method: approle
+          vault-auth-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-auth-secret-id: ${{ secrets.VAULT_SECRET_ID}}
+          vault-url: ${{ secrets.VAULT_ADDR }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.RELEASE_BRANCH }}
+          # Overriding the default GITHUB_TOKEN with a GitHub App token in order to workaround
+          # the known GitHub issue described in https://github.com/camunda/camunda/issues/28522
+          # NOTES:
+          # - This token will be used for all git operations in this job
+          # - This token expires after 1 hour (https://github.com/actions/create-github-app-token?tab=readme-ov-file#create-github-app-token)
+          token: ${{ steps.github-token.outputs.token }}
+
+      - name: Setup Git user
+        run: |
+          git config --global user.email "github-actions[release]"
+          git config --global user.name "github-actions[release]@users.noreply.github.com"
+
+      - name: Set up Java environment
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 21
+          cache: maven
+
+      - name: Update Zeebe version
+        run: |
+          mvn -B versions:set-property -DgenerateBackupPoms=false -Dproperty=dependency.zeebe.version -DnewVersion="${RELEASE_VERSION}"
+          git commit -am "ci: update Zeebe version to ${RELEASE_VERSION}"
+
+      - name: Push changes to release branch
+        if: ${{ inputs.dryRun == false }}
+        run: |
+          git push origin "${RELEASE_BRANCH}"
+
+      - name: Create Git tag
+        run: |
+          git tag -a -m"${RELEASE_VERSION}" "${RELEASE_VERSION}"
+
+      - name: Push Git tag
+        if: ${{ inputs.dryRun == false }}
+        run: |
+          git push origin "${RELEASE_VERSION}"
+
+      - name: Create new stable branch for minor
+        if: ${{ inputs.releaseType == 'minor' }}
+        run: |
+          git checkout --branch "stable/${MINOR_VERSION}"
+          git push origin "stable/${MINOR_VERSION}"
+
+      # Same logic as in camunda/camunda
+      - name: Determine if pre-release
+        id: pre-release
+        run: |
+          shopt -s nocasematch # set matching to case insensitive
+          PRE_RELEASE=false
+          if [[ "${RELEASE_VERSION}" =~ ^.*-(alpha|rc|SNAPSHOT)[0-9]+[\.]*[0-9]*$ ]]; then
```

for snapshots - the version looks like `8.7.0-SNAPSHOT`, right? then it won't be captured (due to the `[0-9]+` part).

```suggestion
          if [[ "${RELEASE_VERSION}" =~ ^.*-(alpha|rc|SNAPSHOT)[0-9]*[\.]*[0-9]*$ ]]; then
```
or something like it.

---

yep, that's fine!

---

actually `.*-SNAPSHOT` would not be a valid release version 🤔 if we do RCs for alphas it would also be rather `*.-alpha2-rc1`. Effectively we could also simply assume any version that has some `-.*` suffix is a pre-release, right?